clients/python/client/osparc/_studies_api.py

```diff
@@ -36,3 +58,53 @@ def create_study_job(self, study_id, job_inputs, **kwargs):
     def clone_study(self, study_id, **kwargs):
         kwargs = {**kwargs, **ParentProjectInfo().model_dump(exclude_none=True)}
         return super().clone_study(study_id, **kwargs)
+
+    def studies(self) -> PaginationGenerator:
+        def pagination_method():
+            page_study = super(StudiesApi, self).list_studies(limit=20, offset=0)
+            assert isinstance(page_study, PageStudy)  # nosec
+            return page_study
+
+        return PaginationGenerator(
+            first_page_callback=pagination_method,
+            api_client=self.api_client,
+            base_url=self.api_client.configuration.host,
+            auth=self._auth,
+        )
+
+    def get_study_job_output_logfiles(self, study_id, job_id) -> Path:
+        return asyncio.run(
+            self.get_study_job_output_logfiles_async(study_id=study_id, job_id=job_id)
+        )
+
+    async def get_study_job_output_logfiles_async(self, study_id, job_id) -> Path:
+        """Download study logs. The log from each node will
+        appear as a file with the node's name in the directory"""
+        logs_map = super().get_study_job_output_logfile(study_id, job_id)
+        assert isinstance(logs_map, JobLogsMap)  # nosec
+        log_links = logs_map.log_links
+        assert log_links  # nosec
+        download_links = [link.download_link for link in log_links]
+        unique_node_names = ensure_unique_names([link.node_name for link in log_links])
+
+        tmp_dir = Path(TemporaryDirectory().name).resolve()
+        tmp_dir.mkdir(parents=True)
+        async with AsyncHttpClient(
+            configuration=self.api_client.configuration
+        ) as client:
+
+            async def _(unique_node_name: str, download_link: str) -> None:
+                response = await client.get(download_link)
+                response.raise_for_status()
+                file = tmp_dir / unique_node_name
+                file.touch()
+                for chunk in response.iter_bytes():
+                    file.write_bytes(chunk)
+
+            tasks = [
+                asyncio.create_task(_(name, link))
+                for name, link in zip(unique_node_names, download_links)
+            ]
+            await asyncio.gather(*tasks)
```

Comment:
NOTE: I would also be careful gathering lists whose length I do not control ... I still did not read @sanderegg links from today but I wonder if something like this might work
```python
import asyncio

async def limited_gather(max_concurrency, *tasks, **kwargs):
    semaphore = asyncio.Semaphore(max_concurrency)
    async def _sem_task(task):
        async with semaphore:
            return await task
    return await asyncio.gather(*(_sem_task(task) for task in tasks), **kwargs)
```

Reply:
I think the function should raise if a download fails. I agree I could try to limit the number of concurrent tasks and that is the proper way to do it. However, here we are talking about number of nodes in a study. So I doubt there will be too many